### ISSUE TITLE
--output ignores configuration

### ISSUE THREAD

**Issue by mtrachy** [conda/conda-build#2165]
Using --output does not work as expected: other command line arguments except the recipe directory are ignored.  It seems that the configuration built from those arguments is not being propagated to `get_output_file_paths()`.

Should [main_build.py:264](https://github.com/conda/conda-build/blob/3.0.3/conda_build/cli/main_build.py#L264) look more like below?

```python
        paths = api.get_output_file_paths(recipe, config=config)
```

**Comment by msarahan** [comment]
Yes, that's a bug. If you have time, please submit a PR so you get credit.
Otherwise I'll fix it very soon.

**Comment by msarahan** [comment]
Fixed by #2174 - thanks again for reporting.
